### DISCUSSION TITLE
Thinning the fat binary to arm64 creates an unopenable application on MacOS

### DISCUSSION THREAD

**Discussion by Red6785** [ghostty-org/ghostty#3139] [Q&A] [answered]
On macOS Sequoia 15.2 arm64, using lipo to thin the fat binary to arm64 creates an unopenable app.
Steps to reproduce:

1. Install Ghostty via the binary.
2. run `lipo /Applications/Ghostty.app/Contents/MacOS/Ghostty -thin arm64 -output /Applications/Ghostty.app/Contents/MacOS/Ghostty`
3. Attempt to boot Ghostty.
Result:
<img width="494" alt="Image" src="https://github.com/user-attachments/assets/f7b061c8-e72b-45a5-9efa-ee477f8409b3" />

**Comment by mitchellh** [comment, marked as answer]
This breaks the code signing and notarization. You cannot do this unless you build from source or sign yourself.

**Reply by Red6785** [reply]
I believe it's possible to sign both architectures individually, isn't it? This is the first app I've installed on my Mac that has been broken via thinning by lipo.

**Reply by mitchellh** [reply]
I'm unsure actually. I thought the notarization would apply to the full app bundle and not just the binaries themselves. If you can find a way to fix this I'd be happy to incorporate it into the release scripts! See this: https://github.com/ghostty-org/ghostty/blob/main/.github/workflows/release-tag.yml

**Reply by Red6785** [reply]
I believe the method is to codesign the two binaries before they are merged via lipo. I am not sure off the top of my head. I can maybe get around to trying soon

**Reply by morajabi** [reply]
FWIW I can confirm thinning by lipo should not break the app, I've done it many times on third party apps.
